lib/oauthservertoserver.js

```
@@ -0,0 +1,92 @@
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+"use strict";
+
+const fetch = require("@adobe/node-fetch-retry");
+const FormData = require('form-data');
+
+/**
+ * Check if an integration configuration is an OAuth Server to Server integration
+ * 
+ * @param {*} integration Integration configuration
+ * @returns {boolean} True if the integration is an OAuth Server to Server integration
+ */
+function isOAuthServerToServerIntegration(integration) {
+    return integration ? integration.TYPE === 'oauthservertoserver' : false;
+}
+
+/**
+ * Validate an OAuth Server to Server integration configuration
+ * 
+ * @param {*} integration Integration configuration
+ * @returns {boolean} True if the integration is a valid OAuth Server to Server integration
+ */
+function validateOAuthServerToServerIntegration(integration) {
+    return integration ? ((typeof integration.ORG_ID === "string") &&
+        Array.isArray(integration.CLIENT_SECRETS) && (integration.CLIENT_SECRETS.length > 0) &&
+        (typeof integration.CLIENT_ID === "string") && 
+        Array.isArray(integration.SCOPES) && (integration.SCOPES.length > 0) &&
+        (typeof integration.TECHNICAL_ACCOUNT_ID === "string") &&
+        (typeof integration.TECHNICAL_ACCOUNT_EMAIL === "string")) : false;
+}
+
+/**
+ * Create an access token for an OAuth Server to Server integration
+ * 
+ * @param {*} integration Integration configuration
+ * @param {string} [adobeLoginHost] IMS host to use, defaults to https://ims-na1.adobelogin.com
+ * @returns {string} access token
+ */
+async function createOAuthServerToServerAccessToken(integration, adobeLoginHost) {
+    // API: https://wiki.corp.adobe.com/display/ims/IMS+API+-+Client+Credentials+Token
+
+    const host = adobeLoginHost || "https://ims-na1.adobelogin.com";
```

would it be better to validate the `adobeLoginHost` ?

---

we could if we want to fail fast, but finally it client configs and wrong config will fail.